core/include/mutex.h

```diff
@@ -253,6 +253,17 @@ static inline void mutex_init(mutex_t *mutex)
     mutex->queue.next = NULL;
 }
 
+/**
+ * @brief   Initializes a mutex object in a locked state.
+ * @details For initialization of variables use MUTEX_INIT_LOCKED instead.
+ *          Only use the function call for dynamically allocated mutexes.
+ * @param[out]      mutex   pre-allocated mutex structure, must not be NULL.
+ */
+static inline void mutex_init_locked(mutex_t *mutex)
+{
+    mutex->queue.next = MUTEX_LOCKED;
```

Comment:
Does this work:

```suggestion
    *mutex = MUTEX_INIT_LOCKED;
```
?

Reply:
That might need a type annotation: `*mutex = (mutex_t)MUTEX_INIT_LOCKED;`

Anyways, doing the same as `mutex_init()` seems better anyways.

Reply:
The benefit of `*mutex = (mutext_t)MUTEX_INIT_LOCKED;` would be that this would also zero-initialize other fields, if they get added. (I think there is a tracing/debug module that adds the program counter of where it was last locked. It would be helpful to not have random values in there.)